### PR TITLE
chore(eslint): migrate eslint configuration .eslintrc.js to eslint.config.mjs

### PR DESCRIPTION
**Pull request by sukvvon**
* [.eslintrc will be deprecated in eslint version 10](https://eslint.org/blog/2023/10/flat-config-rollout-plans/), so i migrate `.eslintrc` to `eslint.config.mjs`

### CODE REVIEW

**Comment by sukvvon** [comment]
@atomiks 

* You will see Tests / Linting and type checking (pull_request) fail, but I only migrated .eslintrc to eslint.config.mjs. The test file already had ESLint errors.
* in turbo, caching might cause tests to appear successful by using cached results from previously successful cases, even if there are actual issues. however, this is not the case in reality.

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    plugins: {
+      react: react,
+      'react-hooks': reactHooks,
+      '@typescript-eslint': tseslint.plugin,
+    },
+    rules: {
+      'no-undef': 'off',
+      'no-unused-expressions': 'off',
+      '@typescript-eslint/no-unused-expressions': 'off',
```
Comment (sukvvon) [inline eslint.config.mjs:32]: * add the ‘no-unused-expressions’ rule to maintain the existing code style

File: extension/src/contexts/devtools.tsx [hunk 2]
```diff
@@ -65,7 +65,6 @@ const DevtoolsContext = React.createContext<DevtoolsContextValue>(
 export const DevtoolsProvider = (
   props: React.ProviderProps<DevtoolsContextValue>,
 ) => (
-  // eslint-disable-next-line react/prop-types
```
Comment (sukvvon) [inline extension/src/contexts/devtools.tsx:68]: * we already 'react/prop-types' rule set off, so i remove eslint-disable-next-line

File: packages/dom/test/visual/spec/Table.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ export function Table() {
               backgroundColor: sameParent === bool ? 'black' : '',
             }}
           >
-            {String(bool) ?? 'None'}
```
Comment (sukvvon) [inline packages/dom/test/visual/spec/Table.tsx:69]: * align format with other components within the 'spec'

File: packages/dom/test/visual/spec/Table.tsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ export function Table() {
               backgroundColor: node === localNode ? 'black' : '',
             }}
           >
-            {localNode ?? 'None'}
```
Comment (sukvvon) [inline packages/dom/test/visual/spec/Table.tsx:85]: * align format with other components within the 'spec'

File: packages/dom/test/visual/spec/Table.tsx [hunk 5]
```diff
@@ -66,7 +66,7 @@ export function Table() {
               backgroundColor: sameParent === bool ? 'black' : '',
             }}
           >
-            {String(bool) ?? 'None'}
+            {String(bool)}
```
Comment (sukvvon) [inline packages/dom/test/visual/spec/Table.tsx:69]: * align format with other components within the `spec`

File: packages/dom/test/visual/spec/Table.tsx [hunk 6]
```diff
@@ -82,7 +82,7 @@ export function Table() {
               backgroundColor: node === localNode ? 'black' : '',
             }}
           >
-            {localNode ?? 'None'}
+            {localNode}
```
Comment (sukvvon) [inline packages/dom/test/visual/spec/Table.tsx:85]: * align format with other components within the `spec`

**Comment by atomiks** [comment]
Thank you!

**Pull request merged**